### PR TITLE
Inteface Created

### LINKED ISSUE #18: Streamlit Interface 🚀📈
**What is Needed**
Contributors are needed to create a very user-friendly interface using streamlit that allows users to predict crime fatalities in Nigeria based on specific input features. The following independent features should be incorporated into the interface:
* year
* source
* source_scale
* location
* interaction
* actor1
* month
* admin2
* admin1
* event_type
* time_precision 
* sub_event_type
* actor2
* inter2
* inter1
* geo_precision
* civilian_targeting
* disorder_type
* notes

**How to Contribute**
1. Access the current script for the [streamlit interface](https://github.com/mlsanigeria/nigeria-crime-trends/blob/main/app.py) and implement your code changes.
2. Test your code: It is pertinent that you check your code runs smoothly before making a pull request.
3. Create a Pull Request: Submit a pull request to the repository with your code in the appropriate script and include any necessary
documentation.

**Getting Started**
Before you begin, ensure you have read the [Contribution Guidelines](https://github.com/mlsanigeria/nigeria-crime-trends/blob/main/CONTRIBUTING.md) in the repository.

We are excited to see your contributions! Happy Hacking! 🎉

### PR DESCRIPTION
**Pull request by hriteshMaikap**
I have createad a simple UI for taking user input. Currently the predict button doesn't work, but inputs can be taken. I have added drop box wherever it was logically feasible based on the categorical values, others are taken in the form of text input. 

https://github.com/user-attachments/assets/935ccf0c-92e8-48f5-a7af-baee1c0919d5

#18

### CODE REVIEW

**Review by Sammybams — CHANGES_REQUESTED** [review]
Thank you very much for your contribution @hriteshMaikap. I would like you to edit the columns to fit in the unique values in each column of the dataset that has been represented. So what you can do is, instead of manually entering the names, reference the file in the data folder and dynamically retrieve all unique values of the columns represented for the dropdowns (categorical columns). And you added a folder `Scripts` which I believe is in error. I already tried deleting the files from it but they are quite much. Please kindly remove it and make the changes requested and this would be good to go. Amazing work Thank you.

**Comment by hriteshMaikap** [comment]
I will look into the issues of scripts folder fs. For the categorical columns, say actor1, it has more than 800 categories. Some even have 2k+ categories (like source), so do you want to include them in the drop down, inspite of thier high count? 
Please let me know, if yes, will include it. 
About Scripts folder, will check and update
image

**Comment by Sammybams** [comment]
> I will look into the issues of scripts folder fs. For the categorical columns, say actor1, it has more than 800 categories. Some even have 2k+ categories (like source), so do you want to include them in the drop down, inspite of thier high count? Please let me know, if yes, will include it. About Scripts folder, will check and update image

Yea, I totally understand this. What I can recommend is to make it dynamic. So to only pick the actor1 for example that have operated in a particular admin 1 and admin2. So that means you must pick admin 1 and admin2 first and then only actor1's in that subset would be displayed.

**Comment by hriteshMaikap** [comment]
Yeah that sounds good to me! Just a small request, if this step of exploring the dataset is already done in any jupyter notebook, could you link it in a reply? It would really help me out and save time!

**Comment by Sammybams** [comment]
> Yeah that sounds good to me! Just a small request, if this step of exploring the dataset is already done in any jupyter notebook, could you link it in a reply? It would really help me out and save time!

You can follow the implementation by @Soumyadeep-Basak [here](https://github.com/mlsanigeria/nigeria-crime-trends/tree/main/experimentation_Soumyadeep_Basak).

**Comment by hriteshMaikap** [comment]
I have added all the inputs dynamically. 
One additional feature is the source column, since there was a combination of input, I manipulate the inputs from the CSV, store it in a set for unqiue values only and allow user to choose mutliple sources, staying true to the input format in the CSV file. 
In case the dropdown is too long, user can type and the most relevant option will automatically be displayed. 
I hope this completes the request :D

File: app.py [hunk 1]
```diff
@@ -1,3 +1,121 @@
 import streamlit as st
-# Set up page configuration
-st.set_page_config(page_title="Nigeira-Crime-Trends", page_icon="📝")
+import pandas as pd
+
+df = pd.read_csv("data\\Nigeria_1997-2024_Sep20.csv")
```
Comment (Sammybams) [inline app.py]:
```suggestion
df = pd.read_csv("./data/Nigeria_1997-2024_Sep20.csv")
```

**Review by Sammybams — APPROVED** [review]
Great work @hriteshMaikap. Thank you very much for your contribution.

**Pull request merged**